backend/utils/flightUtils.js

```diff
@@ -0,0 +1,46 @@
+export function calculateFlightStatus(flightData) {
+    const now = new Date();
+    const flightPoint = flightData.data[0].flightPoints;
+    const departurePoint = flightPoint.find(point => point.departure);
+    const arrivalPoint = flightPoint.find(point => point.arrival);
+    
+    const scheduledDepartureTime = new Date(departurePoint.departure.timings.find(t => t.qualifier === 'STD').value);
+    const scheduledArrivalTime = new Date(arrivalPoint.arrival.timings.find(t => t.qualifier === 'STA').value);
+
+    // Check for delays
+    const departureDelay = getDuration(departurePoint.departure.timings.find(t => t.qualifier === 'STD')?.delays?.[0]?.duration);
+    const arrivalDelay = getDuration(arrivalPoint.arrival.timings.find(t => t.qualifier === 'STA')?.delays?.[0]?.duration);
+
+    if (now < scheduledDepartureTime) {
+        if (departureDelay > 0) {
+            return { status: `Delayed (${formatDuration(departureDelay)})`, departureDelay, arrivalDelay };
+        }
+        return { status: 'Scheduled', departureDelay, arrivalDelay };
+    } else if (now < scheduledArrivalTime) {
+        if (arrivalDelay > 0) {
+            return { status: `In Air - Expected Late (${formatDuration(arrivalDelay)})`, departureDelay, arrivalDelay };
+        }
+        return { status: 'In Air', departureDelay, arrivalDelay };
+    } else {
+        if (arrivalDelay > 0) {
+            return { status: `Arrived Late (${formatDuration(arrivalDelay)})`, departureDelay, arrivalDelay };
```

Comment:
I think putting this logic in the backend is great, but I would think about the goal of this util. What is the utility/reusability of including formatted arrivalDelay in the status string?

Reply:
Basically my idea was to show the delay in the status bar so it's clearer to the user (if there's delay). Do you suggest doing anything else instead?